src/shared/compatibility.js

```diff
@@ -205,23 +205,37 @@ const hasDOM = typeof window === 'object' && typeof document === 'object';
                               globalScope.Promise.prototype.finally)) {
     return;
   }
-  globalScope.Promise = require('core-js/fn/promise');
+  globalScope.Promise = require('core-js/es/promise/index');
+})();
+
+// Support: IE
+(function checkURL() {
+  if (typeof PDFJSDev !== 'undefined' && PDFJSDev.test('IMAGE_DECODERS')) {
+    // The current image decoders don't use the `URL` constructor, so it
+    // doesn't need to be polyfilled for the IMAGE_DECODERS build target.
+    return;
+  }
+  if (typeof PDFJSDev !== 'undefined' && !PDFJSDev.test('GENERIC')) {
+    // The `URL` constructor is assumed to be available in the extension builds.
+    return;
+  }
+  globalScope.URL = require('core-js/web/url');
```

Comment:
Actually, can you please check if *only* `require('core-js/web/url');` is sufficient, without the explicit assignment, since that's probably a better pattern in general here!?